### PR TITLE
Add function to export proxy settings as systemd config

### PR DESCRIPTION
**Pull request by wupeka**
ProxySettings.AsSystemdDefaultEnv exports proxy settings in a form understandable by systemd.

### CODE REVIEW

File: proxy/proxy.go [hunk 1]
```diff
@@ -20,10 +22,11 @@ const (
...
 type Settings struct {
-	Http    string
-	Https   string
-	Ftp     string
-	NoProxy string
+	Http        string
+	Https       string
+	Ftp         string
+	NoProxy     string
+	AutoNoProxy string
```
Comment (jameinel) [inline proxy/proxy.go]: We should probably clearly document AutoNoProxy as it isn't something people who know about proxy stuff are going to immediately understand.

At least an update to the Settings doc.

File: proxy/proxy.go [hunk 2]
```diff
@@ -99,5 +114,18 @@ func (s *Settings) SetEnvironmentValues() {
...
+func (s *Settings) FullNoProxy() string {
+	var allNoProxy []string
+	if s.NoProxy != "" {
+		allNoProxy = strings.Split(s.NoProxy, ",")
+	}
+	if s.AutoNoProxy != "" {
+		allNoProxy = append(allNoProxy, strings.Split(s.AutoNoProxy, ",")...)
+	}
+	noProxySet := set.NewStrings(allNoProxy...)
+	return strings.Join(noProxySet.SortedValues(), ",")
```
Comment (jameinel) [inline proxy/proxy.go]: Ordering doesn't matter, right?

Comment (wupeka) [inline proxy/proxy.go]: It doesn't, it's a set.

**Comment by wupeka** [comment]
!!test!!

**Comment by wupeka** [comment]
$$merge$$

**Comment by wupeka** [comment]
$$merge$$

**Pull request merged**